src/main/java/gttweaker/mods/gregtech/RA2Builder.java

```diff
@@ -174,27 +180,33 @@ public boolean canUndo() {
 
         @Override
         public void undo() {
-            map.mRecipeList.remove(recipe);
-            map.mRecipeItemMap.entrySet()
-                .stream()
-                .filter(
-                    e -> e.getValue()
-                        .removeIf(r -> r == recipe)
-                        && e.getValue()
-                            .size() == 0)
-                .map(Map.Entry::getKey)
-                .collect(Collectors.toSet())
-                .forEach(k -> map.mRecipeItemMap.remove(k));
-            map.mRecipeFluidMap.entrySet()
-                .stream()
-                .filter(
-                    e -> e.getValue()
-                        .removeIf(r -> r == recipe)
-                        && e.getValue()
-                            .size() == 0)
-                .map(Map.Entry::getKey)
-                .collect(Collectors.toSet())
-                .forEach(k -> map.mRecipeFluidMap.remove(k));
+            try {
+                map.mRecipeList.remove(recipe);
+                map.mRecipeItemMap.entrySet()
+                    .stream()
+                    .filter(
+                        e -> e.getValue()
+                            .removeIf(r -> r == recipe)
+                            && e.getValue()
+                                .size() == 0)
+                    .map(Map.Entry::getKey)
+                    .collect(Collectors.toSet())
+                    .forEach(k -> map.mRecipeItemMap.remove(k));
+                map.mRecipeFluidMap.entrySet()
+                    .stream()
+                    .filter(
+                        e -> e.getValue()
+                            .removeIf(r -> r == recipe)
+                            && e.getValue()
+                                .size() == 0)
+                    .map(Map.Entry::getKey)
+                    .collect(Collectors.toSet())
+                    .forEach(k -> map.mRecipeFluidMap.remove(k)); // Sometimes throws java.lang.ClassCastException:
+                                                                  // class
+                                                                  // java.lang.String cannot be cast to class
+                                                                  // net.minecraftforge.fluids.Fluid for some reason
+                                                                  // ?????????
+            } catch (Throwable ignored) {}
```

Comment:
Silently ignoring possible errors seems like a really bad idea, at the bare minimum this should be logged but tbh the underlying crash should be solved

Reply:
Its not crashing in dev with the same exact scripts and required mods for the script to work, only in the modpack.
Doesn't make any sense why there is a string in there, seems very weird.

Reply:
Ok, I found the cause https://github.com/GTNewHorizons/GT5-Unofficial/commit/1338c7d1d6865ef8594eaaa685821014b6fa3fea
Miozune change the key type to String, fixed by updating dependencies.